src/hook/pauseResume.js

```diff
@@ -0,0 +1,318 @@
+			var p;
+			var type = "fx";
+
+			if (getStatus(this) !== "inprogress") {
+				return;
+			}
+
+			//Clear fx-queue except 1 dummy function
+			//for promise not to be expired when calling stop()
+			$.queue(this, type || "fx", [$.noop]);
+			stopFn.call($(this));
+
+			//Remember current animation property
+			if (p = this.__aniProps[0]) {
+				p.elapsed += $.now() - p.start;
+				p.paused = true;
+			}
+		});
+	};
+
+	/**
+	 * Resume animation
+	 * @ko 애니메이션을 재개한다
+	 *
+	 * @name jQuery#resume
+	 * @method
+	 *
+	 * @example
+	 * $("#box").resume(); //resume the paused animation
+	 */
+	$.fn.resume = function() {
+		return this.each(function() {
+			var type = "fx";
+			var p;
+			var i;
+
+			if (getStatus(this) !== "paused") {
+				return;
+			}
+
+			//Clear fx-queue,
+			//And this queue will be initialized by animate call.
+			$.queue(this, type || "fx", []);
+
+			// Restore __aniProps
+			i = 0;
+			while (p = this.__aniProps[i]) {
+				// Restore easing status
+				if (p.elapsed > 0 && p.opt.easing) {
+					var resumePercent = p.elapsed / p.opt.duration;
+					var remainPercent = 1 - resumePercent;
+					var originalEasing = $.easing[p.opt.easing];
+					var startEasingValue = originalEasing(resumePercent);
+					var scale = scaler([startEasingValue, 1], [0, 1]);
+					var newEasingName = p.opt.easing + "_" + p.uuid;
+
+					// Make new easing function that continues from pause point.
+					$.easing[newEasingName] = generateNewEasingFunc(
+						resumePercent, remainPercent, scale, originalEasing);
+					p.opt.easing = newEasingName;
+
+					//Store new easing function to clear it later.
+					p.addEasingFn(newEasingName);
+				}
+
+				p.paused = false;
+				p.opt.duration -= p.elapsed;
+
+				// If duration remains, request 'animate' with storing aniProps
+				if (p.opt.duration > 0 || p.elapsed === 0) {
+					i === 0 && p.init();
+					animateFn.call($(this), p.prop, p.opt);
+				}
+
+				i++;
+			}
+		});
+	};
+
+	/**
+	 * Generate a new easing function.
+	 *
+	 * function to avoid JS Hint error "Don't make functions within a loop"
+	 */
+	function generateNewEasingFunc(resumePercent, remainPercent, scale, originalEasing) {
+		return function easingFunc(percent) {
+			var newPercent = resumePercent + remainPercent * percent;
+			return scale(originalEasing(newPercent));
+		};
+	}
+
+	$.fn.stop = function(type, clearQueue) {
+		var clearQ = clearQueue;
+		stopFn.apply(this, arguments);
+
+		if (typeof type !== "string") {
+			clearQ = type;
+		}
+
+		return this.each(function() {
+			var p;
+
+			if (!clearQ) {
+				p = this.__aniProps.shift();
+				p && p.clearEasingFn();
+			} else {
+				//If clearQueue is requested,
+				//then all properties must be initialized
+				//for element not to be resumed.
+				while (p = this.__aniProps.shift()) {
+					p.clearEasingFn();
+				}
+				this.__aniProps = [];
+			}
+		});
+	};
+
+	jQuery.expr.filters.paused = function(elem) {
+		return getStatus(elem) === "paused" ? true : false;
```

Comment:
Unnecessary conditional ternary operator.
Just it. `return getStatus(elem) === "paused"`